pools-drained/src/agent.ts

```diff
@@ -0,0 +1,106 @@
+import {
+  Finding,
+  TransactionEvent,
+  FindingSeverity,
+  FindingType,
+  Trace,
+  LogDescription,
+  getEthersProvider,
+} from "forta-agent";
+import utils from "./utils";
+import { utils as ethers } from "ethers";
+import abi from "./abi";
+import PairFetcher from "./pair.fetcher";
+
+type Desc = ethers.TransactionDescription;
+
+export const provideHandleTransaction = (fetcher: PairFetcher) => async (txEvent: TransactionEvent) => {
+  const findings: Finding[] = [];
+
+  const details: Record<string, Record<string, number>> = {};
+  const pairs: Set<string> = new Set<string>();
+
+  await Promise.all(
+    txEvent.traces.map(async (trace: Trace) => {
+      const txn = {
+        data: trace.action.input,
+        value: trace.action.value,
+        to: trace.action.to,
+      };
+      try {
+        // is pair V2
+        const desc: Desc = abi.V2_IFACE.parseTransaction(txn);
+        const [valid, token0, token1] = await fetcher.getV2Data(txn.to, txEvent.blockNumber);
+        if (valid && utils.v2Create2(token0, token1) === txn.to) {
+          if (!pairs.has(txn.to)) {
+            pairs.add(txn.to);
+            details[txn.to] = {
+              [token0]: 0,
+              [token1]: 0,
+            };
+          }
+          const calls: number = utils.v2Transfers(desc);
+          details[txn.to][token0] += calls;
+          details[txn.to][token1] += calls;
+        }
+      } catch (e) {}
+      try {
+        // is pair V3
+        const desc: Desc = abi.V3_IFACE.parseTransaction(txn);
+        const [valid, token0, token1, fee] = await fetcher.getV3Data(txn.to, txEvent.blockNumber);
+        if (valid && utils.v3Create2(token0, token1, fee) === txn.to) {
+          if (!pairs.has(txn.to)) {
+            pairs.add(txn.to);
+            details[txn.to] = {
+              [token0]: 0,
+              [token1]: 0,
+            };
+          }
+          const calls: number = utils.v3Transfers(desc);
+          details[txn.to][token0] += calls;
+          details[txn.to][token1] += calls;
+        }
+      } catch (e) {}
+    })
+  );
+
+  const drained: Record<string, boolean> = {};
```

Comment:
As the `drained[i]` never equals to `false`, could this be just a `Set<string>` or a `string[]` for simplicity?

Reply:
Changed to `Set<string>`